sprocs/assessment_instances_select_log.sql

```diff
@@ -342,6 +342,68 @@ BEGIN
                     asl.assessment_instance_id = ai_id
             )
             UNION
+            (
+                SELECT
+                    7.5 AS event_order,
+                    'Time limit expiry'::TEXT AS event_name,
+                    'gray3'::TEXT AS event_color,
+                    asl.date_limit AS date,
+                    u.user_id AS auth_user_id,
+                    u.uid AS auth_user_uid,
+                    NULL::TEXT AS qid,
+                    NULL::INTEGER AS question_id,
+                    NULL::INTEGER AS instance_question_id,
+                    NULL::INTEGER AS variant_id,
+                    NULL::INTEGER AS variant_number,
+                    NULL::INTEGER AS submission_id,
+                    asl.id AS log_id,
+                    jsonb_build_object(
+                         'time_limit', format_interval(asl.date_limit - ai.date)
+                    ) AS data
+                FROM
+                    assessment_state_logs AS asl
+                    JOIN assessment_instances AS ai ON (ai.id = ai_id)
+                    LEFT JOIN users AS u ON (u.user_id = asl.auth_user_id)
+                WHERE
+                    asl.assessment_instance_id = ai_id
+                    AND asl.open
+                    AND asl.date_limit IS NOT NULL
+                    -- Only list as expired if it already happened
+                    AND asl.date_limit < CURRENT_TIMESTAMP
+                    -- Only list the expiry date if the assessment was not closed or extended after this time limit was set but before it expired
+                    AND NOT EXISTS (SELECT 1 FROM assessment_state_logs aslc WHERE aslc.date > asl.date AND aslc.date <= asl.date_limit)
```

Comment:
Do we need a `WHERE` in the `SELECT 1 FROM...` query here? Ditto for the other part of the `UNION` below.

Reply:
Yes. For `aslc.date <= asl.date_limit`, if the assessment is closed or an instructor extends the time limit after the timer expires, we want to see the expiration event in the log. For `aslc.date > asl.date`, if the time limit was modified before this event, it has no impact on the current one.

Reply:
Sorry, I should have been more clear with my question (I agree that the existing `WHERE` clause is necessary): do we need something like `WHERE aslc.assessment_instance_id = ai_id`?

Reply:
Ah, right missed that. Will fix in a minute. Dangers of testing locally...